src/main/java/uk/gov/pay/ledger/queue/EventMessageHandler.java

```diff
@@ -115,7 +128,9 @@ private void processSingleMessage(EventMessage message) throws QueueException {
 
         if (response.isSuccessful()) {
             eventDigestHandler.processEvent(event, response.getState() == INSERTED);
-            eventQueue.markMessageAsProcessed(message);
+            if (message.getQueueMessageReceiptHandle().isPresent()) {
+                eventQueue.markMessageAsProcessed(message);
+            }
```

Comment:
Is an exception woe stopping us from using `message.getQueueMessageReceiptHandle.ifPresent(receiptHandle -> eventQueue.markMessageAsProcessed(message));`?

Reply:
spot on, same issue as above